### PR TITLE
`force=True` is more comprehensive for `powdenest`

### LINKED ISSUE #19627: powdenest(sqrt(sin(x)**2), force=True) does not work
Since `powdenest(sqrt(x**2), force=True)` gives `x`, I would expect `powdenest(sqrt(sin(x)**2), force=True)` to be `sin(x)`.

### PR DESCRIPTION
**Pull request by smichr**
#### References to other Issues or PRs

fixes #19627

#### Brief description of what is fixed or changed

`powdenest` did not always denest exponents; now, except for when the base is negative, the denesting should occur.

#### Release Notes

* simplify
  * use of `force=True` with `powdenest` is more effective

### CODE REVIEW

**Comment by asmeurer** [comment]
> powdenest did not always denest exponents; now, except for when the base is negative, the denesting should occur.

Can you give an example where it explicitly doesn't denest? Would it be sufficient for force=True to replace checks for `is_assumption is True` with `is_assumption is not False`, so that it disallows cases that are known to not work but doesn't fail on unknown assumptions? Although I'm also partial to the idea that it should just always work regardless of assumptions when force=True.

File: sympy/functions/elementary/complexes.py [hunk 1]
```diff
@@ -530,19 +530,27 @@ def eval(cls, arg):
         from sympy.simplify.simplify import signsimp
         from sympy.core.function import expand_mul
         from sympy.core.power import Pow
+        from sympy.functions.elementary.exponential import exp_polar
```
Comment (smichr) [inline sympy/functions/elementary/complexes.py:533]:
```suggestion
```

File: sympy/functions/elementary/complexes.py [hunk 2]
```diff
@@ -530,19 +530,27 @@ def eval(cls, arg):
...
         arg = signsimp(arg, evaluate=False)
         n, d = arg.as_numer_denom()
         if d.free_symbols and not n.free_symbols:
             return cls(n)/cls(d)
 
+        if not arg.atoms(exp_polar):
+            if arg.is_extended_positive:
+                return arg
+            elif arg.is_extended_negative:
+                return -arg
```
Comment (smichr) [inline sympy/functions/elementary/complexes.py:552]:
```suggestion
```

**Comment by smichr** [comment]
> it disallows cases that are known to not work but doesn't fail on unknown assumptions

This is still build in as a safety: if the base is negative then no change will be allowed.

And example from the added tests of something that wouldn't denest before is:
```python
>>> powdenest(sqrt(sin(x)**2), force=True) == sin(x)
True
```
This isn't always negative so we allow the user to make this change. But replacing `sin` with `Function('f', negative=True)` would not denest.

**Pull request merged**